### PR TITLE
[Merged by Bors] - load state in tortoise initializer

### LINKED ISSUE #4140: Node crashes and cannot recover despite restarts
Network 325-short, collector node for explorer ends with forever restart with:
```
{"L":"INFO","T":"2023-03-10T08:22:56.900-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"e7ab103a49","layer_id":6901,"ballot_id":"e7ab103a49","layer_id":6901,"epoch_id":115,"smesher":"2d0d99d0c89d0c8a90387f02663c779fe019b98d7b45ee0f0f54361b761381ee","opinion hash":"0x6a04f71f6cfa1ed1a4fdab70296e960704aabdc6adfef61bbc2ec7057e3ff81d","base_ballot":"df7804774b","support":0,"against":0,"abstain":7,"atx_id":"94ea8f0c2d","ref_ballot":"0000000000","active_set_size":101,"beacon":"7dcd9c9c","votes":{"base":"df7804774b","support":[],"against":[],"abstain":["6894","6895","6896","6897","6898","6899","6900"]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:56.902-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"417a48f9c3","layer_id":6923,"ballot_id":"417a48f9c3","layer_id":6923,"epoch_id":115,"smesher":"c122437381c577eafd588396aac709c4c8147d0503b357bdf90068e0d209b7e7","opinion hash":"0xc502eae1a998dcf104365ed0b461686d2dce5396bd6aed6bc57a207a3188dc0f","base_ballot":"46881cae30","support":0,"against":0,"abstain":7,"atx_id":"6abb305725","ref_ballot":"09dd9dbe09","active_set_size":0,"beacon":"00000000","votes":{"base":"46881cae30","support":[],"against":[],"abstain":["6916","6917","6918","6919","6920","6921","6922"]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:56.926-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"66ba6ddc77","layer_id":6900,"ballot_id":"66ba6ddc77","layer_id":6900,"epoch_id":115,"smesher":"87a36e48f5a3450a06587ed6e254d48f49283df81c1051e52228784d9e3f93e4","opinion hash":"0x6ce25190171d587fd52849e2a88afa58fe5aa616a410cbbab0b294d811dc4fa1","base_ballot":"df7804774b","support":0,"against":0,"abstain":7,"atx_id":"0bca4eb3d4","ref_ballot":"0000000000","active_set_size":101,"beacon":"7dcd9c9c","votes":{"base":"df7804774b","support":[],"against":[],"abstain":["6893","6894","6895","6896","6897","6898","6899"]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:56.935-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"4ecd4420b9","layer_id":6889,"ballot_id":"4ecd4420b9","layer_id":6889,"epoch_id":114,"smesher":"9978b2ed0cd880aafcfbe857d0d40b58e7813449aa277a28afd929a08836dec4","opinion hash":"0x814cb3ab4b962b0170aef06581c7250c86344e20401eaa6de56f42fa7348f63d","base_ballot":"b4c07e0443","support":1,"against":0,"abstain":0,"atx_id":"7115bd4b60","ref_ballot":"6651e87c1e","active_set_size":0,"beacon":"00000000","votes":{"base":"b4c07e0443","support":[{"id":"a7a6ba044a","layer":6888,"height":401092}],"against":[],"abstain":[]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:56.953-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"df7804774b","layer_id":6889,"ballot_id":"df7804774b","layer_id":6889,"epoch_id":114,"smesher":"9a9240f55a5c8da8233ef5b4fdf54b938092929011672a9fbe7fa0218d54a8db","opinion hash":"0x814cb3ab4b962b0170aef06581c7250c86344e20401eaa6de56f42fa7348f63d","base_ballot":"b4c07e0443","support":1,"against":0,"abstain":0,"atx_id":"6bbcd511e2","ref_ballot":"1654cb08e4","active_set_size":0,"beacon":"00000000","votes":{"base":"b4c07e0443","support":[{"id":"a7a6ba044a","layer":6888,"height":401092}],"against":[],"abstain":[]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:56.953-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"c10dc39240","layer_id":6900,"ballot_id":"c10dc39240","layer_id":6900,"epoch_id":115,"smesher":"220fd27814d23e957b4c8d3e94cc51d5835dec44cbb53681a8b2874a553f46cf","opinion hash":"0x6ce25190171d587fd52849e2a88afa58fe5aa616a410cbbab0b294d811dc4fa1","base_ballot":"df7804774b","support":0,"against":0,"abstain":7,"atx_id":"3204967084","ref_ballot":"0000000000","active_set_size":101,"beacon":"7dcd9c9c","votes":{"base":"df7804774b","support":[],"against":[],"abstain":["6893","6894","6895","6896","6897","6898","6899"]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:56.998-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"1bfd7e7728","layer_id":6916,"ballot_id":"1bfd7e7728","layer_id":6916,"epoch_id":115,"smesher":"adbe0b3933d1ff1d3f9114c41810f0c108f38306b68aeb738894b678c595095e","opinion hash":"0x44f85eb492e7b55a1f1ff644f50b964f22560bb06a84f2dfaadea497a115b4f7","base_ballot":"725a4783d2","support":0,"against":0,"abstain":7,"atx_id":"7800a969a4","ref_ballot":"0000000000","active_set_size":101,"beacon":"7dcd9c9c","votes":{"base":"725a4783d2","support":[],"against":[],"abstain":["6909","6910","6911","6912","6913","6914","6915"]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:57.000-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"725a4783d2","layer_id":6889,"ballot_id":"725a4783d2","layer_id":6889,"epoch_id":114,"smesher":"af59f11dd734c1c699bf1c8609a57f55ec32ed5d16dd9be4c59587d38574ad23","opinion hash":"0x814cb3ab4b962b0170aef06581c7250c86344e20401eaa6de56f42fa7348f63d","base_ballot":"b4c07e0443","support":1,"against":0,"abstain":0,"atx_id":"a80a664b89","ref_ballot":"4df73b0635","active_set_size":0,"beacon":"00000000","votes":{"base":"b4c07e0443","support":[{"id":"a7a6ba044a","layer":6888,"height":401092}],"against":[],"abstain":[]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:57.050-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"46881cae30","layer_id":6889,"ballot_id":"46881cae30","layer_id":6889,"epoch_id":114,"smesher":"819ac155c6d2f6d64adaede8861963b39902231efc634dc9cc77048126679d34","opinion hash":"0x814cb3ab4b962b0170aef06581c7250c86344e20401eaa6de56f42fa7348f63d","base_ballot":"b4c07e0443","support":1,"against":0,"abstain":0,"atx_id":"09c89f9302","ref_ballot":"7e9a39f4da","active_set_size":0,"beacon":"00000000","votes":{"base":"b4c07e0443","support":[{"id":"a7a6ba044a","layer":6888,"height":401092}],"against":[],"abstain":[]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:57.053-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"09dd9dbe09","layer_id":6905,"ballot_id":"09dd9dbe09","layer_id":6905,"epoch_id":115,"smesher":"c122437381c577eafd588396aac709c4c8147d0503b357bdf90068e0d209b7e7","opinion hash":"0xe4aaff0927006711aef8736718b6a6bfefd7badffec64fa73c7970786aec0d72","base_ballot":"46881cae30","support":0,"against":0,"abstain":7,"atx_id":"6abb305725","ref_ballot":"0000000000","active_set_size":101,"beacon":"7dcd9c9c","votes":{"base":"46881cae30","support":[],"against":[],"abstain":["6898","6899","6900","6901","6902","6903","6904"]},"name":"proposalListener"}
{"L":"INFO","T":"2023-03-10T08:22:57.053-0500","N":"53a8b.app.53a8b.proposalListener","M":"new ballot","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"app","node_id":"53a8b9642d99d5bf84e5394a25ad9c4dedaefd2c14b9f6a1f2724500e31104dd","module":"proposalListener","sessionId":"fea0c0ab-4d11-43b6-ba0a-3d3910555f10","ballot_id":"5330a4a96a","layer_id":6900,"ballot_id":"5330a4a96a","layer_id":6900,"epoch_id":115,"smesher":"d3ddffddb5ef18fffa46cad45c4e5c28f8bfb9f6c26e23babf45e6a6a5232ed7","opinion hash":"0x6ce25190171d587fd52849e2a88afa58fe5aa616a410cbbab0b294d811dc4fa1","base_ballot":"df7804774b","support":0,"against":0,"abstain":7,"atx_id":"2113c7b910","ref_ballot":"0000000000","active_set_size":101,"beacon":"7dcd9c9c","votes":{"base":"df7804774b","support":[],"against":[],"abstain":["6893","6894","6895","6896","6897","6898","6899"]},"name":"proposalListener"}
fatal error: concurrent map writes

goroutine 278 [running]:
github.com/spacemeshos/go-spacemesh/tortoise.(*state).addBallot(...)
	/src/tortoise/state.go:123
github.com/spacemeshos/go-spacemesh/tortoise.(*turtle).storeBallot(0xc0008a20b0, 0xc004936060)
	/src/tortoise/tortoise.go:791 +0x1da
github.com/spacemeshos/go-spacemesh/tortoise.(*turtle).onBallot(0x1c2c5e0?, 0xc0019a6d40?)
	/src/tortoise/tortoise.go:800 +0x37
github.com/spacemeshos/go-spacemesh/tortoise.(*turtle).loadBallots(0xc0008a20b0, {0x1995ec0?})
	/src/tortoise/tortoise.go:560 +0x130
github.com/spacemeshos/go-spacemesh/tortoise.(*turtle).onLayer(0xc0008a20b0, {0x1c3ded0, 0xc0019a77c0}, {0x19a4a80?})
	/src/tortoise/tortoise.go:360 +0x3c5
github.com/spacemeshos/go-spacemesh/tortoise.New.func1()
	/src/tortoise/algorithm.go:127 +0x85
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
```
full log attached.
[node-crash.txt](https://github.com/spacemeshos/go-spacemesh/files/10942492/node-crash.txt)

Tried restarting node it always ends with that error.

### PR DESCRIPTION
**Pull request by dshulyak**
closes: https://github.com/spacemeshos/go-spacemesh/issues/4140

### CODE REVIEW

**Comment by pigmej** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

File: cmd/node/node.go [hunk 1]
```diff
@@ -526,11 +526,14 @@ func (app *App) initServices(
 	if trtlCfg.BadBeaconVoteDelayLayers == 0 {
 		trtlCfg.BadBeaconVoteDelayLayers = app.Config.LayersPerEpoch
 	}
-	trtl := tortoise.New(app.cachedDB, beaconProtocol,
+	trtl, err := tortoise.New(app.cachedDB, beaconProtocol,
```
Comment (countvonzero) [inline cmd/node/node.go:529]: do you have an idea of how long this would take for the max sliding window size?

**Review by countvonzero — APPROVED** [review]

**Comment by countvonzero** [comment]
i don't fully understand the difference btwn doing the state loading at initialization time vs later before syncer can start.
when the node crash (for other reasons), how does this PR make a difference? 
(other than simplifying code)

**Comment by dshulyak** [comment]
> i don't fully understand the difference btwn doing the state loading at initialization time vs later before syncer can start.
when the node crash (for other reasons), how does this PR make a difference?
(other than simplifying code)

there is a bug in the workflow that i shared in the issue. this change eliminates that bug by removing whole codepath, because the tortoise state will be fully loaded when node joins the network

File: cmd/node/node.go (hunk 1 above)
Comment (dshulyak) [reply cmd/node/node.go:529]: i don't recall, but the duration is the same for previous solution and this one

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**